integration_tests/model_visualization_test.py

```diff
@@ -291,26 +347,40 @@ def call(self, x):
 
 
 def test_plot_nested_functional_model():
-    inputs = keras.Input((3,))
-    x = keras.layers.Dense(4, activation="relu")(inputs)
-    x = keras.layers.Dense(4, activation="relu")(x)
-    outputs = keras.layers.Dense(3, activation="relu")(x)
-    inner_model = keras.Model(inputs, outputs)
-
-    inputs = keras.Input((3,))
-    x = keras.layers.Dense(3, activation="relu", trainable=False)(inputs)
+    inputs = keras.Input((3,), name="input")
+    x = keras.layers.Dense(4, activation="relu", name="dense")(inputs)
+    x = keras.layers.Dense(4, activation="relu", name="dense_1")(x)
+    outputs = keras.layers.Dense(3, activation="relu", name="dense_2")(x)
+    inner_model = keras.Model(inputs, outputs, name="inner_model")
+
+    inputs = keras.Input((3,), name="input_1")
+    x = keras.layers.Dense(
+        3, activation="relu", trainable=False, name="dense_3"
+    )(inputs)
     residual = x
     x = inner_model(x)
-    x += residual
+    x = keras.layers.Add(name="add")([x, residual])
     residual = x
-    x = keras.layers.Dense(4, activation="relu")(x)
-    x = keras.layers.Dense(4, activation="relu")(x)
-    x = keras.layers.Dense(3, activation="relu")(x)
-    x += residual
-    x = keras.layers.Dropout(0.5)(x)
-    outputs = keras.layers.Dense(1, activation="sigmoid")(x)
+    x = keras.layers.Dense(4, activation="relu", name="dense_4")(x)
+    x = keras.layers.Dense(4, activation="relu", name="dense_5")(x)
+    x = keras.layers.Dense(3, activation="relu", name="dense_6")(x)
+    x = keras.layers.Add(name="add_1")([x, residual])
+    x = keras.layers.Dropout(0.5, name="dropout")(x)
+    outputs = keras.layers.Dense(1, activation="sigmoid", name="dense_7")(x)
     model = keras.Model(inputs, outputs)
 
+    edge_dict = get_edge_dict(model_to_dot(model))
+
+    assert edge_dict["input_1 (InputLayer)"] == "dense_3 (Dense)"
```

Comment:
Such statements should be made via `self.assertEqual`

Reply:
I updated the source code as your advice.
Additionally, I removed the main statement also, since, I think it can be replaced by just running ModelVisualizationTest class by pytest.